src/Init/Data/Option/Basic.lean

```diff
@@ -89,6 +89,11 @@ Implementation of `OrElse`'s `<|>` syntax for `Option`.
 instance : OrElse (Option α) where
   orElse := Option.orElse
 
+/-- If the first argument is `some a`, returns `some a`, otherwise returns the second argument. -/
+@[always_inline, inline] def or : Option α → Option α → Option α
```

Comment:
should this be `macro_inline`? That way the second argument will not be evaluated if not needed, similar to `cond`

Reply:
I think if this behavior is desired then `orElse` is a better choice. I have updated the docstring to say as much.

Reply:
I agree, but it still seems unnecessary and inconsistent to have eager evaluation behavior when all other similar functions (like `Bool.or`) have macro inline behavior.

Reply:
I guess we don't lose anything by making it `macro_inline`. Done.